spec/jobs/cron_job/destroy_old_plage_ouverture_job_spec.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+describe CronJob::DestroyOldPlageOuvertureJob do
+  it "Destroy exceptional po closed since 2 months" do
+    now = Time.zone.parse("20220405 10:00")
+    travel_to(now)
+    po_exceptionelle_closed_since_2_months = create(:plage_ouverture, first_day: now - 2.months, recurrence: nil)
```

Comment:
Je ne suis pas sûr de comprendre pourquoi un voyage dans le temps est nécessaire ici, n'aurait-on pas le même résultat avec ceci ?

```suggestion
    po_exceptionelle_closed_since_2_months = create(:plage_ouverture, first_day: 2.months.ago, recurrence: nil)
```

Ça s'applique aussi aux autres exemples. :wink:

Reply:
Pour cette PR Yannick m'a beaucoup aidé (seule je n'aurai pas pu faire ça) dans sa méthode de travail, (si j'ai bien compris sa démarche) il préfère avoir une date fixe de référence d'ou le "now" et le "travel_to"

Reply:
Je trouve que voyager dans le temps permet d'être sur du test, alors que le « from_now » ou des trucs liés à la date courante peut avoir des effets de bord (et donc faire des flaky_test :)).

Reply:
La question est de savoir si l'on veut que ce test nous informe des edge cases en échouant parfois. :wink: En utilisant des dates relatives, on s'assure que le comportement de l'appli est le bon à plusieurs moment, pas à un seul moment. Le test est moins reproductible, mais il teste beaucoup plus de cas (autant qu'il y a d'exécution du test).

Par exemple, sur un autre projet j'ai eu un test qui échouait fin février les années bissextiles, ça m'a permis de corriger un bug qui aurait impacté le comportement de l'app. :exploding_head: 

Je concède que dans le cas présent, la suppression des vieux RDV n'est pas "mission-critical" donc on peut prioriser la stabilité des tests. :+1: 

La concision du code est aussi un autre aspect du sujet : en voyant ce code, je me suis dit qu'il y avait peut-être une raison pour laquelle on voyageait à cette date, autre que la reproductibilité.

Bien sûr, tout ça n'est pas bien important, c'était juste pour contribuer à construire une culture technique avec vous. :pray:

Reply:
Merci d'avoir posé la question, ne serait-ce que pour parler de la culture commune.